### PR TITLE
Support glibc's Safe-Linking in malloc

### PR DESCRIPTION
**Pull request by chkp-eyalit**
### Description/Motivation/Screenshots ###
glibc 2.32 will add support for protection of single-linked-lists such as the Fastbins and TCache lists. This support, which was [pushed](https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=a1a486d70ebcc47a686ff5846875eacad0940e41) already to glibc's master branch, is referred to us "Safe-Linking".

Safe-Linking masks the "fwd" pointers, and so this commit implements the support for unmasking ("reveal"ing) the pointers so they could be used properly by gef.

### How Has This Been Tested? ###

| Architecture |          Yes/No          | Comments                                  |
| ------------ | :----------------------: | ----------------------------------------- |
| x86-32       | :heavy_check_mark: |                                                |
| x86-64       | :heavy_multiplication_x: |                                           |
| ARM          | :heavy_multiplication_x: |                                           |
| AARCH64      | :heavy_multiplication_x: |                                           |
| MIPS         | :heavy_multiplication_x: |                                           |
| POWERPC      | :heavy_multiplication_x: |                                           |
| SPARC        | :heavy_multiplication_x: |                                           |
| RISC-V       | :heavy_multiplication_x: |                                           |
| `make tests` | :heavy_multiplication_x: |                                           |

### Checklist ###

- [X] My PR was done against the `dev` branch, not `master`.
- [X] My code follows the code style of this project.
- [X] My change includes a change to the documentation, if required.
- [X] My change adds tests as appropriate.
- [X] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -787,12 +787,17 @@ def get_next_chunk(self):
...
+        if not sll or get_libc_version() < (2, 32):
+            return read_int_from_memory(self.address)
+        # Unmask ("reveal") the Safe-Linking pointer
+        else:
+            return read_int_from_memory(self.address) ^ (self.address >> 12)
 
     @property
     def fwd(self):
-        return self.get_fwd_ptr()
+        return self.get_fwd_ptr(sll=False)
```
Comment (daniellimws) [inline gef.py:800]: As seen here `sll` will always be false. Could you update the code for fastbins/tcache (all those that use single linked list) to call `self.get_fwd_ptr(true)`?

Comment (chkp-eyalit) [reply gef.py]: Updated as requested. See additional commit below.

**Comment by chkp-eyalit** [comment]
Sending a clean pull request with a single commit. Sorry for the mess

**Comment by Grazfather** [comment]
you could have just squashed and force pushed, or we could have squashed on merge.

**Comment by hugsy** [comment]
See #518 for resolution

**Pull request closed without merging**